wallet-new/src/Cardano/Wallet/Server/Plugins.hs

```diff
@@ -2,267 +2,134 @@
-    -- rewrap some other exceptions (cfr 'rewrapToWalletError').
-    -- Uses the 'Buildable' istance on 'WalletError' to exploit any
-    -- available rendering and information-masking improvements.
-    handleV0Errors :: SomeException -> Maybe Response
-    handleV0Errors se =
-        let maskSensitive err =
-                case err of
-                    V0.RequestError _         -> err
-                    V0.DuplicateWalletError _ -> V0.RequestError "DuplicateWalletError"
-                    V0.NoSuchWalletError _    -> V0.RequestError "NoSuchWalletError"
-                    V0.InternalError _        -> V0.RequestError "InternalError"
-                    V0.DecodeError _          -> V0.RequestError "DecodeError"
-            reify :: V0.WalletError -> V1.WalletError
-            reify = V1.UnknownError . sformat build . maskSensitive
-        in fmap (responseLBS badRequest400 [applicationJson] .  encode . reify) (fromException se)
 
-    -- Handles any generic error, trying to prevent internal exceptions from leak outside.
-    handleGenericError :: SomeException -> Response
-    handleGenericError _ =
-        let
-            unknownV1Error = V1.UnknownError "Something went wrong."
-        in
-            responseLBS badRequest400 [applicationJson] $ encode unknownV1Error
+-- A @Plugin@ running in the monad @m@.
+type Plugin m = [Diffusion m -> m ()]
 
 
--- | A 'Plugin' to start the wallet REST server
---
--- NOTE: There is no web socket support in the new wallet for now.
-walletBackend :: ProtocolMagic
-              -> NewWalletBackendParams
-              -> (PassiveWalletLayer IO, PassiveWallet)
-              -> Plugin Kernel.WalletMode
-walletBackend protocolMagic (NewWalletBackendParams WalletBackendParams{..}) (passiveLayer, passiveWallet) =
+-- | A @Plugin@ to start the wallet REST server
+apiServer
+    :: ProtocolMagic
+    -> NewWalletBackendParams
+    -> (PassiveWalletLayer IO, PassiveWallet)
+    -> Plugin Kernel.WalletMode
+apiServer protocolMagic (NewWalletBackendParams WalletBackendParams{..}) (passiveLayer, passiveWallet) =
     pure $ \diffusion -> do
         env <- ask
         let diffusion' = Kernel.fromDiffusion (lower env) diffusion
         WalletLayer.Kernel.bracketActiveWallet protocolMagic passiveLayer passiveWallet diffusion' $ \active _ -> do
           ctx <- view shutdownContext
-          let
-            portCallback :: Word16 -> IO ()
-            portCallback port = usingLoggerName "NodeIPC" $ flip runReaderT ctx $ startNodeJsIPC port
-          walletServeImpl
+          serveImpl
             (getApplication active)
-            walletAddress
-            -- Disable TLS if in debug modeit .
+            (BS8.unpack ip)
+            port
             (if isDebugMode walletRunMode then Nothing else walletTLSParams)
             Nothing
-            (Just portCallback)
+            (Just $ portCallback ctx)
   where
+    (ip, port) = walletDocAddress
```

Comment:
@KtorZ hehe, just spend a good hour debugging an obscure error which turned out to be this seemingly-innocent binding 😁 

this ought to be `walletAddress` only.

Reply:
Oopsie :no_mouth: